vizro-core/tests/unit/vizro/models/_components/test_tabs.py

```diff
@@ -0,0 +1,74 @@
+"""Unit tests for vizro.models.Container."""
+import dash_mantine_components as dmc
+import pytest
+from asserts import assert_component_equal
+from dash import html
+
+try:
+    from pydantic.v1 import ValidationError
+except ImportError:  # pragma: no cov
+    from pydantic import ValidationError
+
+import vizro.models as vm
+
+
+@pytest.fixture
+def containers():
+    return [
+        vm.Container(id="container-1", title="Title-1", components=[vm.Button()]),
+        vm.Container(id="container-2", title="Title-2", components=[vm.Button()]),
+    ]
+
+
+class TestTabsInstantiation:
+    """Tests model instantiation and the validators run at that time."""
+
+    def test_create_tabs_mandatory_only(self, containers):
+        tabs = vm.Tabs(id="tab-id", tabs=containers)
+        assert isinstance(tabs.tabs[0], vm.Container) and isinstance(tabs.tabs[1], vm.Container)
+        assert tabs.id == "tab-id"
+        assert tabs.type == "tabs"
+
+    def test_mandatory_tabs_missing(self):
+        with pytest.raises(ValidationError, match="field required"):
+            vm.Tabs(id="tab-id")
+
+
+class TestTabsBuildMethod:
```

Comment:
I have been a bit out of the loop with the `assert_component_equal` but this seems quite cumbersome. We repeat the same statement 6 times - this does not seem super intuitive and also not so easy to read what the actual expected outcome of the build method should be.

Reply:
Probably we should prepare an entire expected fixture and write the test as:
```
def test_tabs_build(self, containers, expected):
        result = vm.Tabs(id="tab-id", tabs=containers).build()
        assert_component_equal(result, expected)
```

Reply:
This is mostly due to the fact that putting everything into one `assert_component_equal` wouldn't allow us to test the precision we need e.g. the issue becomes clear when you take a look at the children of this highly nested component tree. I only want to ignore the `children` in the `assert_component_equal` at the lowest level (so ignore the content of the low-level `component.build()`, but I do not want to ignore the children of Tab and Tabslist etc. as these are created in the build)
Splitting it up allows us to test the precision we need (so selectively testing the components/classNames that are actually defined in the `Tabs.build()`.
See this [comment thread](https://github.com/mckinsey/vizro/pull/265#discussion_r1466268165) for reference.

Reply:
I suggested a couple of simplifications here but generally what you wrote was definitely the right strategy @huong-li-nguyen. I agree that a few comments about what the latter assert statements are for would be helpful.

Generally speaking having multiple assert statements in a test is completely fine, so long as it's clear what the point of them is. This is definitely a case where `assert_component_equal` doesn't work so nicely unfortunately so a couple of comments to clarify would be nice.